api/src/main/java/com/velocitypowered/api/command/BrigadierCommand.java

```diff
@@ -0,0 +1,57 @@
+package com.velocitypowered.api.command;
+
+import com.mojang.brigadier.builder.LiteralArgumentBuilder;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.tree.CommandNode;
+import com.mojang.brigadier.tree.LiteralCommandNode;
+import java.util.function.Predicate;
+
+/**
+ * A command that uses Brigadier for parsing the command and providing
+ * suggestions to the client.
+ *
+ * <p>Brigadier commands may be registered using a {@link Builder} instance
+ * obtainable via the {@link CommandManager#brigadierBuilder()} method.
+ */
+public interface BrigadierCommand extends Command {
+
+  /**
+   * Provides a fluent interface to register a Brigadier command.
+   */
+  interface Builder extends Command.Builder<BrigadierCommand, Builder> {
+
+    /**
+     * Specifies the permission-checking predicate the {@link CommandSource}
+     * must pass to execute the command.
```

Comment:
Specify that you should use if you don't want the command to be sent to the backend server at all.

Reply:
My bad, didn't notice this was the previous behavior. Thought `hasPermission` being `false` still returned `true` from `execute` et al. As so, I see this method (and its accompanying implementation) a bit superfluous as the command itself can check the predicate.

Reply:
I'm thinking about the use cases for this, and letting Brigadier commands return a special code (as the implementation does right now) would be simpler and easier (the current injection is a mess).

Reply:
I added a `BrigadierCommand#FORWARD` so implementations can indicate a command execution should be forwarded to the backend server. This is more general and is not only limited to the previous permission-checking logic. Commands can now use the regular `ArgumentBuilder#requires` method to specify a predicate.